File: src/main/java/org/bipmed/brave/variant/Variant.java
package org.bipmed.brave.variant;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Builder;
import lombok.Data;

import java.util.List;

@Data
@Builder
public class Variant {

    @JsonIgnore
    private String id;
    private List<String> snpIds;
    private String datasetId;
    private String assemblyId;
    private Integer totalSamples;
    private String referenceName;
    private Long start;
    private String referenceBases;
    private List<String> alternateBases;
    private List<String> geneSymbol;
    private List<Float> alleleFrequency;
    private Long sampleCount;

    private Statistics coverage;
    private Statistics genotypeQuality;

    private String clnsig;
    private List<String> hgvs;
    private List<String> type;
}

File: src/main/kotlin/org/bipmed/brave/import/variant/Variant.kt
package org.bipmed.brave.import.variant

import com.fasterxml.jackson.annotation.JsonIgnore

data class Variant(
        @JsonIgnore
        val id: String? = null,
        val snpIds: List<String> = emptyList(),
        val datasetId: String? = null,
        val assemblyId: String? = null,
        val totalSamples: Int? = null,
        val referenceName: String? = null,
        val start: Long? = null,
        val referenceBases: String? = null,
        val alternateBases: List<String> = emptyList(),
        val geneSymbol: String? = null,
        val alleleFrequency: List<Number> = emptyList(),
        val sampleCount: Long? = null,

        val coverage: Statistics? = null,
        val genotypeQuality: Statistics? = null,

        val clnsig: String? = null
) {
    data class Statistics(
            val min: Int? = null,
            val q25: Number? = null,
            val median: Number? = null,
            val q75: Number? = null,
            val max: Int? = null,
            val mean: Number? = null
    )
}